admin/class-botamp-admin.php

```diff
@@ -206,59 +130,98 @@ public function api_key_cb() {
 	}
 
 	public function post_type_cb() {
-		$current_post_type = $this->get_option( 'post_type' );
-
-		$html = '<select name = "' . $this->option( 'post_type' ) . '" class = "regular-list" >';
-		foreach ( get_post_types( '', 'objects' ) as $post_type ) {
-			if ( $current_post_type === $post_type->name ) {
-				$html .= "<option value = '{$post_type->name}' selected='true'>{$post_type->label} </option>";
-			} else {
-				$html .= "<option value = '{$post_type->name}'> {$post_type->label} </option>";
-			}
+		$html = '<select class = "botamp-post-type regular-list" >';
+		foreach ( get_post_types( array( 'public' => true ), 'objects' ) as $post_type ) {
+			$html .= "<option value = '{$post_type->name}'> {$post_type->label} </option>";
 		}
 		$html .= '</select>';
-
 		echo $html;
 	}
-
-	public function entity_description_cb() {
-		echo $this->print_field_select( 'entity_description' );
-	}
-
-	public function entity_image_url_cb() {
-		$fields = [ '', 'post_thumbnail_url' ];
-
-		echo $this->print_field_select( 'entity_image_url', $fields );
-	}
-
-	public function entity_title_cb() {
-		echo $this->print_field_select( 'entity_title' );
-	}
-
-	public function entity_url_cb() {
-		echo $this->print_field_select( 'entity_url' );
-	}
-
-	private function print_field_select( $option, $fields = [] ) {
-		$option_value = $this->get_option( $option );
-
-		$fields = empty( $fields ) ? $this->fields : $fields;
-
-		$html = '<select name = "' . $this->option( $option ) . '" class = "regular-list" >';
-		foreach ( $fields as $field ) {
-			if ( $option_value === $field ) {
-				$html .= "<option value = '$field' selected='true'>"
-				. $this->field_name( $field )
-				. '</option>';
-			} else {
-				$html .= "<option value = '$field'>"
-				. $this->field_name( $field )
-				. '</option>';
+	public function entity_fields() {
+		$html = '<div class="botamp-content-mapping">';
+		foreach ( get_post_types( array( 'public' => true ), 'objects' ) as $post_type ) {
+			$option_value = $this->get_option( 'post_type' )[ $post_type->name ];
```

Comment:
`$this->get_option( 'post_type' )` returns a string (see `traits/option.php`). So doing `$this->get_option( 'post_type' )[ $post_type->name ]` doesn't work.